### PR TITLE
add tests to osp-app

### PR DESCRIPTION
**Pull request by Quentinchampenois**
### Add tests workflow to app

Aim is to add tests to osp-app running with github actions and backport custom registration fields

### Tasks
- [x] Add rubocop rules and gem
- [x] Add Github workflows
- [x] Add system and forms tests (not exhaustive list)
- [x] Add i18n spec

### More information
:Deprecated: I plan to open a new PR for fixing CI, the current PR aims to implement a valid CI.

### CODE REVIEW

**Comment by Quentinchampenois** [comment]
@armandfardeau @MoretS @moustachu finally the PR is ready for review

File: spec/i18n_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "i18n/tasks"
+
+describe "I18n sanity" do
+  let(:locales) do
+    ENV["ENFORCED_LOCALES"].presence || "en"
```
Comment (armandfardeau) [inline spec/i18n_spec.rb:7]: What is the current config on the CI? 

We should do en, fr

**Review by armandfardeau — APPROVED** [review]

File: spec/i18n_spec.rb (hunk 1 above)
Comment (Quentinchampenois) [reply spec/i18n_spec.rb:7]: At the moment there is no configuration on the CI for the language, but I can add a configuration and update locales to `en`, `fr` in `spec/spec_helper.rb`. What do you think about this suggestion ?

Comment (armandfardeau) [reply spec/i18n_spec.rb]: Perfect

**Pull request merged**